tests/total_zhat_jet.py

```diff
@@ -0,0 +1,33 @@
+import jax.numpy as np
+from jax import jvp, jet, grad
+
+
+def f(z, t):
+  return 4 * z / t
+
+
+# Initial Conditions
+t0, z0 = 2., 0.5
+
+
+# True solution
+def sol(t):
+  return (z0 / t0**4) * t**4
+
+
+# Derivatives at t_eval
+t_eval = 4.
+z_eval = sol(t_eval)
+
+# True total derivatives
+dz = grad(sol)(t_eval)
+ddz = grad(grad(sol))(t_eval)
+dddz = grad(grad(grad(sol)))(t_eval)
+
+# Jet total derivatives
+z_terms = (np.ones_like(z_eval),np.zeros_like(z_eval))
+t_terms = (np.ones_like(t_eval),np.zeros_like(t_eval))
+
+jet(f,(z_eval,t_eval),np.array([z_terms,t_terms]))
```

Comment:
I can explain this on call, but essentially `sol` provides the closed form solution for `sol(t) = f(z(t),t)` such that `grad` gives the "total derivative" wrt t.

However, we need this total derivative without directly having access to `z(t)`, instead we have access to `z_eval` 

This test will eventually help us determine how to use `jet` on `z_eval` to get the "total derivative" wrt z of `f(z_eval, t_eval)` 

However, this is stopped currently by bugs in my implementation of `mul` and the symbolic zeros.